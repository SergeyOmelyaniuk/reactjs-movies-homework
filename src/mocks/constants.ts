import { Category, Genre, Language } from '../types';

export const categories: Category[] = [
	{ id: 1, value: 'popular', title: 'Popular' },
	{ id: 2, value: 'top_rated', title: 'Top rated' },
	{ id: 3, value: 'upcoming', title: 'Upcoming' },
];

export const languages: Language[] = [
	{ title: 'en', value: 'en-US' },
	{ title: 'ru', value: 'ru-RU' },
];

export const genres: Genre[] = [
	{ id: 1, name: 'Genre 1' },
	{ id: 2, name: 'Genre 2' },
	{ id: 3, name: 'Genre 3' },
	{ id: 4, name: 'Genre 4' },
	{ id: 5, name: 'Genre 5' },
];
