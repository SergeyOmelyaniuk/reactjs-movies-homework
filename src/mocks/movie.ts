import { MovieDetails } from './../types';

export const movieDescription: MovieDetails = {
	poster_path: 'film_1.jpg',
	genres: [{ id: 1, name: 'first genre' }],
	title: 'Красотка на взводе',
	release_date: '2021-05-26',
	vote_average: 68,
	revenue: 120000,
	runtime: 120,
	overview:
		'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
};
