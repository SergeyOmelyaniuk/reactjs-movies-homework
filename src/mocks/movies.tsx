import { Movie } from '../types';

export const movies: Movie[] = [
	{
		id: 1,
		title: 'Красотка на взводе',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_1.jpg',
		vote_average: 68,
		popularity: 0.5,
		revenue: 120000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [
			{
				name: 'Kate Beckinsale',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_1.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Lindy',
			},
			{
				name: 'Stanley Tucci',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_2.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Dr. Ivan Munchin',
			},
			{
				name: 'Bobby Cannavale',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_3.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Detective Vicars',
			},
			{
				name: 'Jai Courtney',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_4.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Justin',
			},
			{
				name: 'Laverne Cox',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_5.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Detective Nevin',
			},
			{
				name: 'David Bradley',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_6.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Gareth Fizel',
			},
		],
	},
	{
		id: 2,
		title: 'Сокровище Счастливчика',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_2.jpg',
		vote_average: 72,
		popularity: 0.5,
		revenue: 230000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 3,
		title: 'Сталинград',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_3.jpg',
		vote_average: 76,
		popularity: 0.5,
		revenue: 240000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 4,
		title: 'Люди в черном: Интернэшнл',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_4.jpg',
		vote_average: 69,
		popularity: 0.5,
		revenue: 136000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 5,
		title: 'Значит, война',
		release_date: '2021-05-26',
		genre: ['Боевик'],
		img: 'film_5.jpg',
		vote_average: 96,
		popularity: 0.5,
		revenue: 226000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
];
