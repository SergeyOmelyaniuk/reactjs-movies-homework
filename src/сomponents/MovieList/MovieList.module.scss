@import '../../scss/variables.scss';

.title {
	font-size: 25px;
	margin-bottom: 30px;
}

.noResults {
	font-size: 30px;
	text-transform: uppercase;
	color: $colorError;
}
