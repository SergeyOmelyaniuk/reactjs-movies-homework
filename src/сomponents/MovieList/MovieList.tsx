import React from 'react';
import { Movie } from '../../types';
import WrapMovies from '../WrapMovies';
import styles from './MovieList.module.scss';

interface MovieListProps {
	movies: Movie[];
}

function MovieList(props: MovieListProps) {
	return (
		<div>
			<h3 className={styles.title}>Search results:</h3>
			{props.movies.length === 0 ? (
				<p className={styles.noResults}>No results found</p>
			) : (
				<WrapMovies movies={props.movies} />
			)}
		</div>
	);
}

export default MovieList;
