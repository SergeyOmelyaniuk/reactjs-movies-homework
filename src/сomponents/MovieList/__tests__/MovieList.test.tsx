import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import MovieList from '..';
import { movies } from '../../../mocks/movies';

describe('MovieList', () => {
	test('should render with search results', () => {
		const { container } = render(
			<BrowserRouter>
				<MovieList movies={movies.popular} />
			</BrowserRouter>
		);

		expect(container).toMatchSnapshot();
	});

	test('should render when search results is empty', () => {
		const { container } = render(
			<BrowserRouter>
				<MovieList movies={[]} />
			</BrowserRouter>
		);

		expect(container).toMatchSnapshot();
	});
});
