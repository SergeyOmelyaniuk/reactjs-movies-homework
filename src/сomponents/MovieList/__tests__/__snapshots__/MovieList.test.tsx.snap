// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList should render when search results is empty 1`] = `
<div>
  <div>
    <h3
      class="title"
    >
      Search results:
    </h3>
    <p
      class="noResults"
    >
      No results found
    </p>
  </div>
</div>
`;

exports[`MovieList should render with search results 1`] = `
<div>
  <div>
    <h3
      class="title"
    >
      Search results:
    </h3>
    <div
      class="wrapMovies"
    >
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/1"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_1.jpg"
            />
            <span
              class="rating"
            >
              68.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Красотка на взводе
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
              <span>
                Genre 2
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/2"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_2.jpg"
            />
            <span
              class="rating"
            >
              72.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Сокровище Счастливчика
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
              <span>
                Genre 2
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/3"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_3.jpg"
            />
            <span
              class="rating"
            >
              76.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Сталинград
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/4"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film42.jpg"
            />
            <span
              class="rating"
            >
              69.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Люди в черном: Интернэшнл
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 4
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/5"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_5.jpg"
            />
            <span
              class="rating"
            >
              96.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Значит, война
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/6"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_6.jpg"
            />
            <span
              class="rating"
            >
              77.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Приключения желтого чемоданчика
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/7"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_7.jpg"
            />
            <span
              class="rating"
            >
              79.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Поезд в Пусан
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/8"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_8.jpg"
            />
            <span
              class="rating"
            >
              79.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Никто не выжил
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 3
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/9"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_9.jpg"
            />
            <span
              class="rating"
            >
              53.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Ведьма
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/10"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_10.jpg"
            />
            <span
              class="rating"
            >
              53.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Звёздные войны: Пробуждение силы
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/11"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_11.jpg"
            />
            <span
              class="rating"
            >
              58.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Марсианин
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/12"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_12jpg"
            />
            <span
              class="rating"
            >
              100.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Республика Z
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 3
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/13"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_13.jpg"
            />
            <span
              class="rating"
            >
              90.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Планета 51
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/14"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_14.jpg"
            />
            <span
              class="rating"
            >
              86.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Красные туфельки и семь гномов
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/15"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_15.jpg"
            />
            <span
              class="rating"
            >
              92.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Капитошка
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/16"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_16.jpg"
            />
            <span
              class="rating"
            >
              58.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Война будущего
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/17"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_17.jpg"
            />
            <span
              class="rating"
            >
              86.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Чернобыль
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 5
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/18"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_18.jpg"
            />
            <span
              class="rating"
            >
              81.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Странный дом
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 3
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/19"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_19.jpg"
            />
            <span
              class="rating"
            >
              79.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Дальний космос
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
            </p>
          </div>
        </a>
      </div>
      <div
        class="wrapMovie"
      >
        <a
          class="movieCard"
          href="/movies/20"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt="movie"
              class="image"
              src="https://image.tmdb.org/t/p/w500/film_20.jpg"
            />
            <span
              class="rating"
            >
              20.0
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Мандалорец
            </h2>
            <p
              class="genre"
            >
              <span>
                Genre 1
              </span>
            </p>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
